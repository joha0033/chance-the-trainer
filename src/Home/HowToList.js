import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';

const styles = theme => ({
  root: {
    width: '100%',
    maxWidth: 450,
    backgroundColor: theme.palette.background.paper,
  },
});

function ListItemLink(props) {
  return <ListItem  {...props} />;
}

function HowToList(props) {
  const { classes } = props;

  return (
    <div className={classes.root} >
      <List component="nav">
        <h3>How this works</h3>
        <ListItem>
          <ListItemText primary=" This is an app that helps with Public Speaking." />
        </ListItem>
        <ListItemLink href="#simple-list">
          <ListItemText primary="Its a random slide show generator" />
        </ListItemLink>
        <ListItemLink href="#simple-list">
          <ListItemText primary="3 slides (photos) // 30 seconds a piece" />
        </ListItemLink>
        <ListItemLink href="#simple-list">
          <ListItemText primary="Start a speech inspired by the photo displayed" />
        </ListItemLink>
        <ListItemLink href="#simple-list">
          <ListItemText primary="Relate your points from previous slide with the next" />
        </ListItemLink>
        <ListItemLink href="#simple-list">
          <ListItemText primary="Tie your points together with the last slide" />
        </ListItemLink>
        <ListItemLink href="#simple-list">
          <ListItemText primary="Click the get started button" />
        </ListItemLink>
        <ListItemLink href="#simple-list">
          <ListItemText primary="Once directed to Slideshow, press the play button" />
        </ListItemLink>
      </List>
    </div>
  );
}

HowToList.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(HowToList);