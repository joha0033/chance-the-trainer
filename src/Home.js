import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import HowToList from './HowToList'
import GetStartedButton from './GetStartedButton';

const styles = theme => ({
  root: {
    ...theme.mixins.gutters(),
    marginBottom: 30,
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2,
    textAlign: 'center'
  }
});

function Home(props) {
  const { classes } = props;
  return (
    <div>
      <Paper  className={classes.root} elevation={5}>
        <Typography variant="h5" component="h4" >
          <HowToList />
          <GetStartedButton />
        </Typography>
      </Paper>
    </div>
  );
}

Home.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Home);