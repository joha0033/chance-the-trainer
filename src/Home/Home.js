import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import HowToList from './HowToList'
import GetStartedButton from './GetStartedButton';
import withWidth from '@material-ui/core/withWidth';

const styles = theme => ({
  root: {
    ...theme.mixins.gutters(),
    marginBottom: 30,
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2,
    textAlign: 'center',
  }
});

function Home(props) {
  const { classes, width } = props;
  const elevation = width === 'xs' ? 0 : 5
  
  return (
        <Paper  className={classes.root} elevation={elevation}>
          <Typography component="h4" >
            <HowToList />
            <GetStartedButton />
          </Typography>
        </Paper>
  );
}

Home.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(withWidth()(Home));