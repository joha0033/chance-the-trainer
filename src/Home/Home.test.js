import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {render, fireEvent, cleanup, text} from 'react-testing-library';
import { MemoryRouter } from 'react-router-dom'
import Home from './Home';
import HowToList from './HowToList'
import GetStartedButton from './GetStartedButton'

/**
 * BASIC TESTS
 * 
 * Renders, Snapshots
 * 
 * START
 */

 /**
  * RENDERS START
  */
it('renders Home component without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <Home />
    </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders get started button without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <GetStartedButton />
    </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders how to list without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<HowToList />, div);
  ReactDOM.unmountComponentAtNode(div);
});
 /**
  * RENDERS END
  */


/**
 * SNAPSHOTS START
 */
it('renders entire Home component correctly', () => {
  const tree = renderer
    .create(<MemoryRouter>
        <Home />
      </MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// HOW TO LIST
it('renders how to list correctly', () => {
  const tree = renderer
    .create(<HowToList />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// GET STARTED BUTTON
it('renders get started button correctly', () => {
  const tree = renderer
    .create(
    <MemoryRouter>
      <GetStartedButton />
    </MemoryRouter>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
/**
 * SNAPSHOTS END
 * 
 * BASIC TESTS END
 */


 /**
  * DOM && EVENTS
  * 
  * START
  */
 afterEach(cleanup);

 it('Home display elevation variable changes with screen width', () => {
   const propXS = {
    width: 'xs'
   }
   const propsSM = {
    width: 'sm'
   }
  const componentLarge = renderer
    .create(
    <MemoryRouter>
        <Home  {...propsSM} />
    </MemoryRouter>)
    .toJSON()
  const componentSmall = renderer
    .create(
    <MemoryRouter>
       <Home  {...propXS} />
    </MemoryRouter>
     )
    .toJSON()
    
  expect(componentSmall.children[0].props.className.includes('MuiPaper-elevation0-224')).toBeTruthy();
  expect(componentLarge.children[0].props.className.includes('MuiPaper-elevation5-229')).toBeTruthy();

});
 /**
  * END
  * 
  * DOM && EVENTS
  */
