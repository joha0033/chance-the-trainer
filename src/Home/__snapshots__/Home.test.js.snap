// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders entire Home component correctly 1`] = `
<div>
  <div
    className="MuiPaper-root-222 MuiPaper-elevation5-229 MuiPaper-rounded-223 WithTheme-WithWidth-Home---root-221"
  >
    <h4
      className="MuiTypography-root-249 MuiTypography-body1-258"
    >
      <div
        className="HowToList-root-285"
      >
        <nav
          className="MuiList-root-286 MuiList-padding-287"
        >
          <h3>
            How this works
          </h3>
          <li
            className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
            disabled={false}
          >
            <div
              className="MuiListItemText-root-302"
            >
              <span
                className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
              >
                 This is an app that helps with Public Speaking.
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
            disabled={false}
            href="#simple-list"
          >
            <div
              className="MuiListItemText-root-302"
            >
              <span
                className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
              >
                Its a random slide show generator
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
            disabled={false}
            href="#simple-list"
          >
            <div
              className="MuiListItemText-root-302"
            >
              <span
                className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
              >
                3 slides (photos) // 30 seconds a piece
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
            disabled={false}
            href="#simple-list"
          >
            <div
              className="MuiListItemText-root-302"
            >
              <span
                className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
              >
                Start a speech inspired by the photo displayed
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
            disabled={false}
            href="#simple-list"
          >
            <div
              className="MuiListItemText-root-302"
            >
              <span
                className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
              >
                Relate your points from previous slide with the next
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
            disabled={false}
            href="#simple-list"
          >
            <div
              className="MuiListItemText-root-302"
            >
              <span
                className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
              >
                Tie your points together with the last slide
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
            disabled={false}
            href="#simple-list"
          >
            <div
              className="MuiListItemText-root-302"
            >
              <span
                className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
              >
                Click the get started button
              </span>
            </div>
          </li>
          <li
            className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
            disabled={false}
            href="#simple-list"
          >
            <div
              className="MuiListItemText-root-302"
            >
              <span
                className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
              >
                Once directed to Slideshow, press the play button
              </span>
            </div>
          </li>
        </nav>
      </div>
      <div>
        <a
          href="/slideshow"
          onClick={[Function]}
        >
          <button
            className="MuiButtonBase-root-335 MuiButton-root-309 MuiButton-contained-320 MuiButton-containedPrimary-321 MuiButton-raised-323 MuiButton-raisedPrimary-324 GetStartedButton-button-308"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-310"
            >
              Get Started
            </span>
            <span
              className="MuiTouchRipple-root-338"
            />
          </button>
        </a>
      </div>
    </h4>
  </div>
</div>
`;

exports[`renders get started button correctly 1`] = `
<div>
  <a
    href="/slideshow"
    onClick={[Function]}
  >
    <button
      className="MuiButtonBase-root-335 MuiButton-root-309 MuiButton-contained-320 MuiButton-containedPrimary-321 MuiButton-raised-323 MuiButton-raisedPrimary-324 GetStartedButton-button-308"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-310"
      >
        Get Started
      </span>
      <span
        className="MuiTouchRipple-root-338"
      />
    </button>
  </a>
</div>
`;

exports[`renders how to list correctly 1`] = `
<div
  className="HowToList-root-285"
>
  <nav
    className="MuiList-root-286 MuiList-padding-287"
  >
    <h3>
      How this works
    </h3>
    <li
      className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
      disabled={false}
    >
      <div
        className="MuiListItemText-root-302"
      >
        <span
          className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
        >
           This is an app that helps with Public Speaking.
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
      disabled={false}
      href="#simple-list"
    >
      <div
        className="MuiListItemText-root-302"
      >
        <span
          className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
        >
          Its a random slide show generator
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
      disabled={false}
      href="#simple-list"
    >
      <div
        className="MuiListItemText-root-302"
      >
        <span
          className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
        >
          3 slides (photos) // 30 seconds a piece
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
      disabled={false}
      href="#simple-list"
    >
      <div
        className="MuiListItemText-root-302"
      >
        <span
          className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
        >
          Start a speech inspired by the photo displayed
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
      disabled={false}
      href="#simple-list"
    >
      <div
        className="MuiListItemText-root-302"
      >
        <span
          className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
        >
          Relate your points from previous slide with the next
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
      disabled={false}
      href="#simple-list"
    >
      <div
        className="MuiListItemText-root-302"
      >
        <span
          className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
        >
          Tie your points together with the last slide
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
      disabled={false}
      href="#simple-list"
    >
      <div
        className="MuiListItemText-root-302"
      >
        <span
          className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
        >
          Click the get started button
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-root-290 MuiListItem-default-293 MuiListItem-gutters-298"
      disabled={false}
      href="#simple-list"
    >
      <div
        className="MuiListItemText-root-302"
      >
        <span
          className="MuiTypography-root-249 MuiTypography-subheading-256 MuiListItemText-primary-305"
        >
          Once directed to Slideshow, press the play button
        </span>
      </div>
    </li>
  </nav>
</div>
`;
